src/ChannelServer/World/SpawnZone.cs

```diff
@@ -0,0 +1,219 @@
+				return;
+			}
+			Log.Debug("map {0}", _map.Name);
+
+			_isInitialized = false;
+			_countDeaths = 0;
+
+			_spawnEntities = new Dictionary<int, Monster>();
+			rndGenerator = new Random();
+		}
+
+		public void StartRespawn()
+		{
+			if (!_isInitialized)
+			{
+				Log.Warning("Spawn {0} {1} can't start respawn because its not initialized.", this.Id, this.Data.spawnName);
+				return;
+
+			}
+
+			_respawnEnabled = true;
+		}
+
+		public void StopRespawn()
+		{
+			_respawnEnabled = false;
+		}
+
+		public void Init()
+		{
+			if (_map == null || this.MonsterData == null)
+				return;
+
+			// Calculate count of spawn entities
+			Random rnd = new Random();
+			_calculatedTotalEntities = this.Data.count + rnd.Next(this.Data.countVariation);
+			_countEntities = 0;
+
+			_isInitialized = true;
+			_respawnEnabled = true;
+
+			for (int i = 0; i < _calculatedTotalEntities; i++)
+			{
+				DoSpawn(null);
+			}
+
+		}
+
+		public void DoSpawn(Object obj)
+		{
+			Log.Debug("DoSpawn");
+			if (!_isInitialized && !_respawnEnabled)
+				return;
+
+
+			Monster monster = new Monster(this.MonsterData.Id, Shared.Const.NpcType.Monster);
+			if (monster == null)
+				return;
+
+			Log.Debug("New monster created {0}", monster.Handle);
+
+			monster = InitializeMonster(monster);
+
+			Log.Debug("Monster initialized");
+
+			lock (_spawnEntities)
+			{
+				_spawnEntities.Add(monster.Handle, monster);
+				_countEntities = _spawnEntities.Count;
+			}
+
+			Log.Debug("Add monster {0}", monster.Handle);
+			_map.AddMonster(monster);
+
+		}
+
+		public Monster InitializeMonster(Monster monster)
+		{
+			if (monster == null)
+				return null;
+
+			// Position
+			if (this.Data.isFixedLocation)
+			{
+				monster.Position = new Position(this.Data.Position.X, this.Data.Position.Y, this.Data.Position.Z);
+			}
+			else
+			{
+				// Search a random location inside the polygon
+				Position rndPosition;
+				if (!this.GetRandomPosition(out rndPosition))
+				{
+					Log.Warning("There was not possible to find a random position in polygon in Spawn {0} {1}", this.Id, this.Data.spawnName);
+					return null;
+				}
+				monster.Position = rndPosition;
+			}
+
+			//monster.AI = new AIMonster(monster);
+			//monster.AI.SetIntention(IntentionTypes.AI_INTENTION_ACTIVE);
+
+			// Make a reference to this spawn zone in the monster.
+			monster.spawnZone = this;
+
+			return monster;
+		}
+
+		public bool GetRandomPosition(out Position rndPosition)
+		{
+			int xPos, zPos;
+			for (int i = 0; i < 100; i++)
+			{
+				xPos = rndGenerator.Next(this.Data.xMin, this.Data.xMax);
+				zPos = rndGenerator.Next(this.Data.zMin, this.Data.zMax);
+
+				if (IsInPolygon(xPos, zPos))
+				{
+					rndPosition = new Position(xPos, 260, zPos);
+					return true;
+				}
+
+				if (i == 50)
+				{
+					Log.Warning("Polygon seems complicated in spawn {0} {1}", this.Id, this.Data.spawnName);
```

Comment:
It looks you're using a simple rectangle for the spawn area, there shouldn't be a reason to use loops or have warnings for complicated polygons, because you will definitely find a random position in a rectangle. There's not even a need for the "IsInside" check.